lib/streamlit/elements/pyplot.py

```diff
@@ -32,8 +34,9 @@ def pyplot(dg, fig=None, clear_figure=None, **kwargs):
         Parameters
         ----------
         fig : Matplotlib Figure
-            The figure to plot. When this argument isn't specified, which is
-            the usual case, this function will render the global plot.
+            The figure to plot. When this argument isn't specified, this
```

Comment:
I'm wondering if this deprecation (and deprecations in general) should have a nice visible INFO callout (the way we do for beta_ features?)

Pros: More immediately clear, user will see warning before the issue instead of after tripping over it.
Cons: More work to add and remove, makes our docs even longer than they are already, it's an easy fix once they've tripped

Thoughts? (I expect Thiago will want to workshop this docstring, so this might be more of a question for him)

Reply:
@tvst ^

Reply:
I don't understand what you mean by an "INFO callout".

But in terms of workshopping this docstring, I think something like this would be clearer:

```
            The figure to plot. When this argument isn't specified, this function will render the global figure (but this is deprecated, as described below)
```
... then later on:

```
.. info :: Deprecation warning. We will soon remove the ability to specify no arguments in `st.plot()`, as that requires the use of Matplotlib's global figure object, which is not thread-safe. So please always pass a figure object as shown in the example section below.
```